scylla/src/transport/locator/mod.rs

```diff
@@ -549,3 +553,240 @@ impl<'a> Iterator for ReplicaSetIterator<'a> {
         }
     }
 }
+
+impl<'a> ReplicaSet<'a> {
+    pub(crate) fn into_replicas_ordered(self) -> ReplicasOrdered<'a> {
```

Comment:
> To accomplish that, the (abstractly unordered)
> ReplicaSet becomes convertible into a newly defined ReplicasOrdered
> struct. Iterating over ReplicasOrdered yields replicas in ring order, so
> policies can depend on it to perform LWT optimisation correctly.

If other policies can use it (I presume that you meant also user-defined ones), then why is everything `pub(crate)`?

Reply:
This is my general bias towards restricting everything to `pub(crate)` - this time it was wrong. I'll change it to `pub`.